src/chain/tezos/contracts/HicNFT.ts

```diff
@@ -0,0 +1,232 @@
+            operatorsMapId: Number(JSONPath({ path: '$.args[1].args[1].int', json: storageResult })[0]),
+            paused: (JSONPath({ path: '$.args[2].prim', json: storageResult })[0]).toString().toLowerCase().startsWith('t'),
+            tokenMetadataMapId: Number(JSONPath({ path: '$.args[3].int', json: storageResult })[0])
+        };
+    }
+
+    /*
+     * hDao FA2 contract storage type
+     */
+    export type HDaoStorage = HicStorage;
+
+
+    /*
+     * Get an instance of an hDao contract's storage by querying a given address
+     * @param server The Tezos node to communicate with
+     * @param address The deployed Timelock contract address
+     */
+    export async function getHDaoStorage(server: string): Promise<HDaoStorage> {
+        const storageResult = await TezosNodeReader.getContractStorage(server, hDaoAddress);
+        return {
+            administrator: JSONPath({path: '$.args[0].args[0].string', json: storageResult })[0],
+            allTokens: Number(JSONPath({ path: '$.args[0].args[1].int', json: storageResult })[0]),
+            ledgerMapId: Number(JSONPath({ path: '$.args[0].args[2].int', json: storageResult })[0]),
+            metadataMapId: Number(JSONPath({ path: '$.args[1].args[0].int', json: storageResult })[0]),
+            operatorsMapId: Number(JSONPath({ path: '$.args[1].args[1].int', json: storageResult })[0]),
+            paused: (JSONPath({ path: '$.args[2].prim', json: storageResult })[0]).toString().toLowerCase().startsWith('t'),
+            tokenMetadataMapId: Number(JSONPath({ path: '$.args[3].int', json: storageResult })[0])
+        };
+    }
+
+    /*
+     * FA2 individual token transfer
+     */
+    interface TokenTransaction {
+        destination: string;
+        token_id: number;
+        amount: number;
+    }
+
+    /*
+     * Returns a TokenTransaction in Michelson format
+     */
+    export function TokenTransactionMichelson(tx: TokenTransaction): string {
+        return `Pair "${tx.destination}" (Pair ${tx.token_id} ${tx.amount})`;
+    }
+
+    /*
+     * FA2 batch transfer invocation parameters
+     */
+    interface TransferPair {
+        source: string;
+        txs: TokenTransaction[]
+    }
+
+    /*
+     * Returns a TransferPair in Michelson format
+     */
+    export function TransferPairMichelson(transfers: TransferPair[]): string {
+        const transferList: string = transfers.map(
+            (transfer) => {
+                const txList: string = transfer.txs.map((tx) => TokenTransactionMichelson(tx)).join("; ");
+                return `Pair "${transfer.source}" { ${txList} }`;
+            }
+        ).join("; ");
+        return `{ ${transferList} }`
+    }
+
+    /*
+     * hDao transaction parameters
+     */
+    export type HDaoTransaction = TokenTransaction;
+
+    /*
+     * hDao transfer entrypoint parameters
+     */
+    export type HDaoTransferPair = TransferPair;
```

Comment:
@itamarreif: seems like a redundant way to expose the interfaces, also reuse [MultiAssetTokenHelper](https://github.com/Cryptonomic/ConseilJS/blob/master/src/chain/tezos/contracts/tzip12/MultiAssetTokenHelper.ts) instead.